MyCryptoHQ/MyCrypto
# Handle MetaMask network change

Closes #1966

### Description

Because MetaMask has chosen to stop reloading pages on network change, we need to implement our own solution when a change is detected.

See: https://medium.com/metamask/breaking-change-no-longer-reloading-pages-on-network-change-4a3e1fd2f5e7

### Changes

* Once per second, MetaMask's network is compared to the local version.
* If the versions don't match, the user will be logged out and a notification alerting them to the change will appear.

### Steps to Test

1. Access a wallet using MetaMask.
2. Switch the network in the MetaMask browser extension.
3. The wallet should be reset and you should be logged out.

**NOTE: This is going to be hard to test while the extension still refreshes the page, as it seems to take precedence over everything else. I tested this "in code" by changing window.web3 to return an arbitrary chain ID, but I'm not sure how we can test this before the change to the extension.**
